go-controller/pkg/ovn/namespace.go

```diff
@@ -244,48 +250,49 @@ func (oc *Controller) updateNamespace(old, newer *kapi.Namespace) {
 	annotation = newer.Annotations[routingExternalGWsAnnotation]
 	oldAnnotation = old.Annotations[routingExternalGWsAnnotation]
 	if annotation != oldAnnotation {
-		var stderr string
-		var err error
-		for podIP, gwToGr := range nsInfo.podExternalRoutes {
-			for gw, gr := range gwToGr {
-				mask := GetIPFullMask(podIP)
-				_, stderr, err = util.RunOVNNbctl("--", "--if-exists", "--policy=src-ip",
-					"lr-route-del", gr, podIP+mask, gw)
+		// if old gw annotation was empty, new one must not be empty, so we should remove any per pod SNAT
+		if oldAnnotation == "" {
+			if config.Gateway.DisableSNATMultipleGWs && (len(nsInfo.routingExternalGWs) != 0 || len(nsInfo.routingExternalPodGWs) != 0) {
```

Comment:
maybe a helper function or two for this logic?

and/or move more of this function into egressgw.go?

(maybe later)

Reply:
this logic is only done here, so I think its ok to leave it for now